src/helpers/debounce.test.ts

```diff
@@ -0,0 +1,155 @@
+import GovukButton from "govuk-frontend/govuk/components/button/button";
+import React from "react";
+
+import { debounce, makeDebounceContext } from "./debounce";
+
+it("returns `false` and doesn't prevent the default action on both of two call within the default timeout without `data-prevent-double-click` set on the event's target, similar to `GovukButton.prototype.debounce`", () => {
+  expect.hasAssertions();
+
+  const mockPreventDefault = jest.fn();
+  const event = {
+    ...jest.fn()(),
+    target: {
+      ...jest.fn()(),
+      getAttribute(qualifiedName: string): string | null {
+        if (qualifiedName === "data-prevent-double-click") {
+          return null;
+        }
+
+        return "attribute-value";
+      }
+    } as HTMLElement,
+    preventDefault: mockPreventDefault
+  } as React.MouseEvent<HTMLElement>;
+
+  const runTestFor = (
+    method: (event: React.MouseEvent<HTMLElement>) => boolean | void,
+    notDebouncedResult: boolean | undefined
+  ): void => {
+    jest.runAllTimers();
+    mockPreventDefault.mockReset();
+
+    expect(method(event)).toEqual(notDebouncedResult);
+
+    expect(mockPreventDefault).not.toHaveBeenCalled();
+
+    jest.advanceTimersByTime(999);
+
+    expect(method(event)).toEqual(notDebouncedResult);
+
+    expect(mockPreventDefault).not.toHaveBeenCalled();
+  };
+
+  {
+    const context = makeDebounceContext();
+
+    runTestFor(event => debounce(event, context), false);
+  }
+
+  {
+    const context = {};
+
+    runTestFor(GovukButton.prototype.debounce.bind(context), undefined);
+  }
+});
+
+it('returns `true` and prevents the default action of the event when called a second time within the default timeout with `data-prevent-double-click` set to `"true"` on the event\'s target, similar to `GovukButton.prototype.debounce`', () => {
```

Comment:
"returns `true` and prevents the default action of the event on a second call within the default timeout when `data-prevent-double-click` set to `"true"` on the event\'s target, similar to `GovukButton.prototype.debounce`'